core/scripts/i18n/collect-strings.js

```diff
@@ -613,43 +614,100 @@ function writeStringsToCtcFiles(locale, strings) {
 }
 
 /**
- * This function does two things:
- *
- *    - Add `meaning` property to ctc messages that have the same message but different descriptions so TC can disambiguate.
- *    - Throw if the known collisions has changed at all.
+ * TODO: replace by Array.prototype.groupToMap when available.
+ * @template {unknown} T
+ * @template {string} U
+ * @param {Array<T>} elements
+ * @param {(element: T) => U} callback
+ * @return {Map<U, Array<T>>}
+ */
+function arrayGroupToMap(elements, callback) {
```

Comment:
Had to do this twice (nested) and was getting kind of gross. It's [a nice API](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/groupToMap) (for Node 20 or whatever)